packages/core/admin/admin/src/content-manager/hooks/useContentManagerInitData.ts

```diff
@@ -30,53 +31,79 @@ interface ContentManagerLink {
   isDisplayed: boolean;
 }
 
-/**
- * TODO: refactor this whole thing.
- */
 const useContentManagerInitData = () => {
   const dispatch = useTypedDispatch();
   const toggleNotification = useNotification();
-  const state = useTypedSelector((state) => state['content-manager_app']);
-
   const { allPermissions } = useRBACProvider();
   const { runHookWaterfall } = useStrapiApp();
   const { notifyStatus } = useNotifyAT();
   const { formatMessage } = useIntl();
   const { get } = useFetchClient();
+  const state = useTypedSelector((state) => state['content-manager_app']);
 
-  const fetchDataRef = useRef(async () => {
-    dispatch({
-      type: 'ContentManager/App/GET_INIT_DATA',
-    });
-
-    try {
-      const {
-        data: {
-          data: { components, contentTypes: models, fieldSizes },
-        },
-      } = await get<Contracts.Init.GetInitData.Response>('/content-manager/init');
-      notifyStatus(
-        formatMessage({
-          id: getTranslation('App.schemas.data-loaded'),
-          defaultMessage: 'The schemas have been successfully loaded.',
-        })
-      );
-
-      const {
-        data: { data: contentTypeConfigurations },
-      } = await get<Contracts.ContentTypes.FindContentTypesSettings.Response>(
-        '/content-manager/content-types-settings'
-      );
+  const fetchInitialData = async () => {
+    const {
+      data: {
+        data: { components, contentTypes, fieldSizes },
+      },
+    } = await get<Contracts.Init.GetInitData.Response>('/content-manager/init');
+
+    return { components, contentTypes, fieldSizes };
+  };
+
+  const fetchContentTypeSettings = async () => {
+    const {
+      data: { data: contentTypeConfigurations },
+    } = await get<Contracts.ContentTypes.FindContentTypesSettings.Response>(
+      '/content-manager/content-types-settings'
+    );
+
+    return contentTypeConfigurations;
+  };
+
+  const handleError = (error: any) => {
+    console.error(error);
+    toggleNotification({ type: 'warning', message: { id: 'notification.error' } });
```

Comment:
i'd probably assert that this is an AxiosError and use the helper hook to correctly parse the message and show that in the banner and if it's not then show the generic one? Also probably no point in console erroring lol